repose-aggregator/core/valve/src/test/scala/org/openrepose/spring/ValveTestModeRunnerTest.scala

```diff
@@ -0,0 +1,415 @@
+import org.junit.runner.RunWith
+import org.openrepose.commons.config.manager.UpdateListener
+import org.openrepose.core.container.config.ContainerConfiguration
+import org.openrepose.core.systemmodel.SystemModel
+import org.openrepose.valve.jmx.ValvePortMXBean
+import org.openrepose.valve.spring.ValveRunner
+import org.scalatest.concurrent.Eventually
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.{FunSpec, Matchers}
+import org.slf4j.LoggerFactory
+
+import scala.concurrent.{Await, Future}
+
+
+@RunWith(classOf[JUnitRunner])
+class ValveTestModeRunnerTest extends FunSpec with Matchers with LazyLogging with Eventually {
+  val log = LoggerFactory.getLogger(this.getClass)
+
+  val fakeConfigService = new FakeConfigService()
+
+  import scala.concurrent.ExecutionContext.Implicits.global
+  import scala.concurrent.duration._
+
+  /**
+   * For this class, the test mode is going to be true!
+   * @param configRoot
+   * @param insecure
+   * @param testMode
+   * @param f
+   * @return
+   */
+  def withRunner(configRoot: String = "/config/root",
+                 insecure: Boolean = false,
+                 testMode: Boolean = true)(f: ValveRunner => Unit) = {
+    val runner = new ValveRunner(fakeConfigService)
+    val runnerTask = Future {
+      runner.run(configRoot, insecure, testMode)
+    }
+
+    runnerTask.onFailure {
+      case t => fail("Future didn't go!", t)
+    }
+    try {
+      f(runner)
+    } finally {
+      runner.destroy()
+    }
+    Await.ready(runnerTask, 3 seconds)
+  }
+
+  def getValvePortMXBean: ValvePortMXBean = {
+    val mbs = ManagementFactory.getPlatformMBeanServer
+    val name = new ObjectName(ValvePortMXBean.OBJECT_NAME)
+    if (mbs.isRegistered(name)) {
+      println("Getting the registered mbean!")
+      val thing = JMX.newMBeanProxy(mbs, name, classOf[ValvePortMXBean])
+      thing
+    } else {
+      fail("Unable to get ValvePort MX Bean!")
+    }
+  }
+
+  def updateSystemModel(resource: String): UpdateListener[SystemModel] = {
+    val systemModelListener = fakeConfigService.getListener[SystemModel]("system-model.cfg.xml")
+    val systemModel = Marshaller.systemModel(resource)
+    systemModelListener.configurationUpdated(systemModel)
+    systemModelListener
+  }
+
+  def updateContainerConfig(resource: String): UpdateListener[ContainerConfiguration] = {
+    val containerListener = fakeConfigService.getListener[ContainerConfiguration]("container.cfg.xml")
+    val containerConfig = Marshaller.containerConfig(resource)
+    containerListener.configurationUpdated(containerConfig)
+    containerListener
+  }
+
+  it("has a blocking run method") {
+    val runner = new ValveRunner(fakeConfigService)
+
+    val future = Future {
+      runner.run("/root", false)
+    }
+    future.isCompleted shouldBe false
+
+    runner.destroy()
+    Await.ready(future, 1 second)
+    future.isCompleted shouldBe true
+  }
+
+  it("passes through configRoot and insecure to each node") {
+    pending
```

Comment:
Yeah I need to delete that test. There's several that were copied over that I never ended up implenting that don't really have value. 

On May 8, 2015 3:01:46 PM CDT, Adrian J George III <notifications@github.com> wrote:
>> +
>> +  it("has a blocking run method") {
>> +    val runner = new ValveRunner(fakeConfigService)
>> +
>> +    val future = Future {
>> +      runner.run("/root", false)
>> +    }
>> +    future.isCompleted shouldBe false
>> +
>> +    runner.destroy()
>> +    Await.ready(future, 1 second)
>> +    future.isCompleted shouldBe true
>> +  }
>> +
>> +  it("passes through configRoot and insecure to each node") {
>> +    pending
>
>?
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/rackerlabs/repose/pull/1297/files#r29971691

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.